code/controllers/QueuedJobsDevAdmin.php

```diff
@@ -0,0 +1,276 @@
+			}
+		}
+	}
+
+	public function job($request) {
+		$classSegment = $request->param('JobClassName');
+		$class = str_replace('-', '\\', $classSegment);
+		if (!$class) {
+			return $this->echoTemplate(array(
+				'Content' => 'Cannot provide blank class name',
+			));
+		}
+		if (!class_exists($class)) {
+			// NOTE(Jake): Would be nice to make a "best" guess here or add some
+			//			   sort of partial match execute for command-line UX
+			return $this->echoTemplate(array(
+				'Content' => 'Class "'.$class.'" does not exist.',
+			));
+		}
+		$list = $this->getJobList()->filter(array('Implementation' => $class));
+
+		$actionOrID = $request->param('ID');
+		if (!$actionOrID) {
+			// View job list
+			$segment = $classSegment.'/any';
+			$jobs = array(
+				array(
+					'Title' => 'Execute "any" job (creates and executes a job if none exist)',
+					'Description' => 'Will execute any item on this list, if none exist, this will create a new job of this type and immediately execute.',
+					'Segment' => $segment,
+					'Link' => $this->Link($segment),
+				)
+			);
+			foreach ($list->sort('ID') as $jobRecord) {
+				$segment = $classSegment.'/'.$jobRecord->ID;
+				// Show step status if it's used by the job
+				$stepStatus = '';
+				if ($jobRecord->TotalSteps) {
+					$stepStatus = ' (Step: '.(int)$this->StepsProcessed.'/'.(int)$jobRecord->TotalSteps.')';
+				}
+				$jobs[] = array(
+					'Title' => 'Execute #'.$jobRecord->ID.' '.$jobRecord->Implementation,
+					'Description' => $jobRecord->getTitle().$stepStatus,
+					'Segment' => $segment,
+					'Link' => $this->Link($segment),
+				);
+			}
+			if (!Director::is_cli()) {
+				return $this->echoTemplate(array(
+					'Records' => new ArrayList($jobs),
+				), 'QueuedJobsDevAdmin_recordlist');
+			}
+			foreach($jobs as $job) {
+				echo " * ".$job['Title']." : sake dev/jobs/" . $job['Segment'] . "\n";
+			}
+			return;
+		}
+		if (is_numeric($actionOrID)) {
+			$job = $list->byID($actionOrID);
+			if (!$job || !$job->exists()) {
+				return $this->echoTemplate(array(
+					'Content' => 'Cannot find job by ID #'.$actionOrID.' of class "'.$class.'"'
+				));
+			}
+			if (Director::is_cli()) {
+				echo "Starting existing job #$job->ID $job->Implementation \n";
+			}
+			return $this->executeJobAndRender($job->ID, 
+					'Successfully executed existing job #{$ID} "{$Class}"',
+					'Broken job #{$ID} "{$Class}"');
+		}
+		switch ($actionOrID) {
+			case "any":
+				$job = $list->sort('ID')->first();
+				if ($job) {
+					if (Director::is_cli()) {
+						echo "Starting existing job #$job->ID $job->Implementation \n";
+					}
+					return $this->executeJobAndRender($job->ID, 
+					'Successfully ran existing job #{$ID} "{$Class}"',
+					'Broken existing job #{$ID} "{$Class}"');
+				}
+				// If no items exist, create one with default parameters and execute.
+				$job = method_exists($class, 'create') ? $class::create() : new $class;
```

Comment:
Is it possible to provide a list of constructor args to be used in this circumstance? Many jobs types expect constructor params for initialisation, which this doesn't cater for. 

Also, can you check that $class implements QueuedJob before creating an instance? Rather than an arbitrary URL-supplied class being created here.